### PR TITLE
[RSDK-4101] Update API for monitoring SLAM session from job and job checkout

### PR DESCRIPTION
**Pull request by nroede**
Add changes from the [scope doc](https://docs.google.com/document/d/1hsXdESoLryw1c7-AiAbtsBqdegy2O0Sb2SmLl6imHhU/edit) to implement cloud run job monitoring and updating of the MDB doc through app, as well as a few other minor changes.

Ticket: https://viam.atlassian.net/browse/RSDK-4101

### CODE REVIEW

**Comment by tessavitabile** [comment]
Thanks! I will merge once everyone has approved.

File: proto/viam/app/cloudslam/v1/cloud_slam.proto [hunk 1]
```diff
@@ -74,14 +77,37 @@ message ListMappingSessionsResponse {
...
+// UpdateMappingSessionMetadataByID
+
+message UpdateMappingSessionMetadataByIDRequest {
+  string session_id = 1;
+  string end_status = 2;
+}
+
+
+message UpdateMappingSessionMetadataByIDResponse {
+  bool success = 1;
```
[the remark this answers is not in the document]
Comment (nroede) [reply proto/viam/app/cloudslam/v1/cloud_slam.proto]: yes, somehow slipped through. corrected now!

File: proto/viam/app/cloudslam/v1/cloud_slam.proto [hunk 2]
```diff
@@ -74,14 +77,33 @@ message ListMappingSessionsResponse {
...
+
+message GetMappingSessionMetadataByIDResponse {
+  MappingMetadata session_metadata = 1;
+}
+
+// UpdateMappingSessionMetadataByID
+
+message UpdateMappingSessionMetadataByIDRequest {
+  string session_id = 1;
+  string end_status = 2;
```
Comment (JohnN193) [inline proto/viam/app/cloudslam/v1/cloud_slam.proto:101]: [nit] add a comment for end status, similar to the one in MappingMetadata

**Review by JohnN193 — APPROVED** [review]
LGTM, just a small nit

**Review by benjirewis — APPROVED** [review]

**Review by EshaMaharishi — APPROVED** [review]

File: proto/viam/app/cloudslam/v1/cloud_slam.proto (hunk 2 above)
Comment (nroede) [reply proto/viam/app/cloudslam/v1/cloud_slam.proto:102]: done!

**Comment by nroede** [comment]
> Is this ready to merge?

Yes, thank you!

**Pull request merged**